src/bulk_insert/bulk_insert.c

```diff
@@ -216,6 +216,7 @@ int BulkInsert(RedisModuleCtx *ctx, GraphContext *gc, RedisModuleString **argv,
 
 	// lock graph under write lock
 	// allocate space for new nodes and edges
+	Graph_SetMatrixPolicy(g, RESIZE_TO_CAPACITY);
```

Comment:
I think that at the end of this function, we should have a call that reverts this change:
```
Graph_SetMatrixPolicy(g, SYNC_AND_MINIMIZE_SPACE);
```
Otherwise, we risk finishing the bulk insert and leaving the graph with the non-standard policy.

Reply:
absolutely right!